src/emulator/controller.ts

```diff
@@ -225,6 +225,65 @@ export function shouldStart(options: any, name: Emulators): boolean {
   return emulatorInTargets;
 }
 
+function findExportMetadata(importPath: string): ExportMetadata | undefined {
+  const pathIsDirectory = fs.lstatSync(importPath).isDirectory();
+  if (!pathIsDirectory) {
+    return;
+  }
+
+  // If there is an export metadata file, we always prefer that
+  const importFilePath = path.join(importPath, HubExport.METADATA_FILE_NAME);
+  if (fileExistsSync(importFilePath)) {
+    return JSON.parse(fs.readFileSync(importFilePath, "utf8").toString()) as ExportMetadata;
+  }
+
+  const fileList = fs.readdirSync(importPath);
+
+  // The user might have passed a Firestore export directly
+  const firestoreMetadataFile = fileList.find((f) => f.endsWith(".overall_export_metadata"));
+  if (firestoreMetadataFile) {
+    const metadata: ExportMetadata = {
+      version: EmulatorHub.CLI_VERSION,
+      firestore: {
+        version: "prod",
+        path: importPath,
+        metadata_file: `${importPath}/${firestoreMetadataFile}`,
+      },
+    };
+
+    EmulatorLogger.forEmulator(Emulators.FIRESTORE).logLabeled(
+      "BULLET",
+      "firestore",
+      `Detected non-emulator Firestore export, creating metadata file ${importFilePath}`
+    );
+    fs.writeFileSync(importFilePath, JSON.stringify(metadata, undefined, 2));
+
+    return metadata;
+  }
+
+  // The user might haved passed a directory containing RTDB json files
+  const rtdbDataFile = fileList.find((f) => f.endsWith(".json"));
```

Comment:
This may introduce many false positives (including `firebase.json`) and I'd suggest we reconsider this. Plus, how many developers will actually take advantage of this feature? I don't think any production service or tool right now create directories that contain all RTDB namespace exports besides the CLI itself. And for a single JSON file, the Emulator UI drag-and-drop experience should already serve as a stop-gap solution.

Reply:
So I agree false positives are an issue, I tried to cut down on that by making sure they pass a directory (so to hit `firebase.json` you'd have to pass `--import=./` which shouldn't look right to anyone).

I have two reasons for wanting this to stay:

- I want import/export to work for all emulators in all cases wherever possible to reduce cognitive load on developers.
- If you *do* want to import multiple namespaces, this is flexible and powerful in a way that drag-and-drop is not.

We will definitely have to document this feature though.

Reply:
Okay, it's great that we had this discussion and let's proceed as you said.